code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -598,13 +598,12 @@
 // and due to an oversight in turf/Enter() were going through walls.  That
 // should be independently resolved, but this is also an interesting twist.
 /obj/structure/closet/Exit(atom/movable/AM)
-	open()
+	open(AM)
```

Comment:
I don't know what's going on here. @Kurfursten

Reply:
Exit() is a built in proc to BYOND (you can see the API by opening the code and pressing F1) - it has an excellent description there.

This is a good example of why we want to use user not usr, because this Exit() would be called if an object leaves the closet, and it would force the closet (or bodybag by extension) open. An example of how this might happen is if it were recalled by bound object, moved by an admin, beepsky trying to chase a criminal, etc. In this case it probably would make open() runtime before because the usr might not make sense.

This seems fine, and then inside your bodybag code you can just add a check `if(!istype(user)) return 1` before the do_after call. This would mean objects can't open the locker if someone is standing up inside, which seems fine to me.